docs/source/integrations/spark.rst

```diff
@@ -59,26 +65,37 @@ DataFrame containing the data from the file.
 .. code-block:: python
 
     from pyspark.sql.types import StructType, StructField, FloatType, StringType
-    schema = SparkStruct(
-        StructField(name=”sepal_length”, FloatType(), False),
-        StructField(name=”sepal_width”, FloatType(), False),
-        StructField(name=”petal_length”, FloatType(), False),
-        StructField(name=”petal_width”, FloatType(), False),
-    )
-    df = spark.read.csv("https://docs.bentoml.org/en/latest/_static/examples/batch/input.csv")
+    import urllib.request
+
+    urllib.request.urlretrieve("https://docs.bentoml.org/en/latest/_static/examples/batch/input.csv", "input.csv")
+
+    schema = StructType([
+        StructField("sepal_length", FloatType(), False),
+        StructField("sepal_width", FloatType(), False),
+        StructField("petal_length", FloatType(), False),
+        StructField("petal_width", FloatType(), False),
+    ])
+    df = spark.read.csv("input.csv", schema=schema)
 
 Create a BentoService object
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Create a BentoService object using the BentoML service you want to use for the batch inference
-job. You can do this by calling the ``bentoml.get`` function, and passing the name of the bento
-and its version as a parameter.
+job. Here, we first try to use ``bentoml.get`` to get the bento from the local BentoML store. If it
+is not found, we retrieve the bento from the BentoML public S3 and import it.
 
 .. code-block:: python
 
     import bentoml
 
-    bento = bentoml.get("iris_classifier:latest")
+    try:
+        bento = bentoml.get("iris_classifier:atfmp3u3ncrkseb5")
+    except bentoml.exceptions.NotFound:
+        import urllib.request
+        urllib.request.urlretrieve("https://bentoml-public.s3.us-west-1.amazonaws.com/quickstart/iris_classifier.bento", "iris_classifier.bento")
```

Comment:
This is necessary if we don't want to require users to setup S3 credentials.

Reply:
I think this complicates the guide. We can mentioned in the prerequisite that, this guide assumes you have completed the tutorial and has an iris_classifier bento ready in the bento store. If not, you can run the following to import the quickstart bento.
```
import urllib.request
        urllib.request.urlretrieve("https://bentoml-public.s3.us-west-1.amazonaws.com/quickstart/iris_classifier.bento", "iris_classifier.bento")
bentoml.import_bento("iris_classifier.bento")
```